ts/Series/Treegraph/TreegraphSeries.ts

```diff
@@ -0,0 +1,659 @@
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    public init(): void {
+        super.init.apply(this, arguments);
+        if (H.treeLayouts[this.options.layout]) {
+            this.layoutAlgorythm = new H.treeLayouts[this.options.layout]();
+        } else {
+            this.layoutAlgorythm = new H.treeLayouts.Walker();
+        }
+    }
+
+    /**
+     * Extend generatePoints by adding the nodes, which are Point objects
+     * but pushed to the this.nodes array.
+     * @private
+     */
+    public generatePoints(): void {
+        NodesComposition.generatePoints.apply(this, arguments as any);
+
+        /**
+         * Order the nodes, starting with the root node(s). (#9818)
+         * @private
+         */
+        function order(node: TreegraphNode, level: number): void {
+            // Prevents circular dependencies in the data:
+            if (node.wasVisited) {
+                throw Error('circular dependency deteceted');
+            } else {
+                level = node.level || level;
+                node.level = level;
+                node.wasVisited = true;
+                node.linksFrom.forEach(function (link: TreegraphPoint): void {
+                    if (link.toNode) {
+                        order(link.toNode, level + 1);
+                    }
+                });
+            }
+        }
+
+        if (this.orderNodes) {
+            this.nodes
+                // Identify the root node(s)
+                .filter(function (node: TreegraphNode): boolean {
+                    return node.linksTo.length === 0;
+                })
+                // Start by the root node(s) and recursively set the level
+                // on all following nodes.
+                .forEach(function (node: TreegraphNode): void {
+                    order(node, 0);
+                });
+            stableSort(
+                this.nodes,
+                function (a: TreegraphPoint, b: TreegraphPoint): number {
+                    return a.level - b.level;
+                }
+            );
+        }
+        // To enable tooltip for link
+        this.data.forEach(function (link): void {
+            link.formatPrefix = 'link';
+        });
+    }
+
+    private getLayoutModifiers(): LayoutAlgorythmParameters {
+        const chart = this.chart,
+            plotSizeX = chart.plotSizeX as number,
+            plotSizeY = chart.plotSizeY as number;
+        let minX = Infinity,
+            maxX = -Infinity,
+            minY = Infinity,
+            maxY = -Infinity,
+            maxXSize = 0,
+            minXSize = 0,
+            maxYSize = 0,
+            minYSize = 0;
+        this.nodes.forEach((node: TreegraphNode): void => {
+            const markerOptions = merge(
+                this.options.marker,
+                node.options.marker
+            );
+            const symbol = markerOptions.symbol;
+            node.symbol = symbol;
```

Comment:
Why are we adding it to this place? Looks to not be related to the method

Reply:
I guess it is not necessary, it can be read straight from the options when it is necessary.